plugins/outputs/newrelic/newrelic_test.go

```diff
@@ -0,0 +1,134 @@
+package newrelic
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+	"github.com/stretchr/testify/require"
+)
+
+func ErrorContains(out error, want string) bool {
+	if out == nil {
+		return want == ""
+	}
+	if want == "" {
+		return false
+	}
+	return strings.Contains(out.Error(), want)
+}
+
+func TestBasic(t *testing.T) {
+	nr := &NewRelic{
+		MetricPrefix: "Test",
+		InsightsKey:  "12345",
+		Timeout:      internal.Duration{Duration: time.Second * 5},
+	}
+
+	err := nr.Connect()
+	require.NoError(t, err)
+
+	err = nr.Write(testutil.MockMetrics())
```

Comment:
This test makes a connection to newrelics servers, so it would need to be skipped on -short.  However the big issue here is that this test doesn't test anything specifically.

Reply:
Skipped the test case, asserted the error to contain a particular string, this is more of a negative test case